src/main/java/org/itxtech/nemisys/network/RakNetInterface.java

```diff
@@ -111,12 +111,12 @@ public void emergencyShutdown() {
 
     @Override
     public void openSession(String identifier, String address, int port, long clientID) {
-        PlayerCreationEvent ev = new PlayerCreationEvent(this, Player.class, Player.class, null, address, port);
+        PlayerCreationEvent ev = new PlayerCreationEvent(this, Player.class, Player.class, clientID, address, port);
         this.server.getPluginManager().callEvent(ev);
         Class<? extends Player> clazz = ev.getPlayerClass();
 
         try {
-            Constructor constructor = clazz.getConstructor(SourceInterface.class, Long.class, String.class, int.class);
+            Constructor constructor = clazz.getConstructor(SourceInterface.class, long.class, String.class, int.class);
```

Comment:
long.class, hm? it's not Long?

Reply:
This is not causing the error

Reply:
If it was the case then `int.class` would be `Integer.class`